Gems/AudioSystem/Code/Source/Editor/AudioControlsEditorUndo.cpp

```diff
@@ -273,6 +273,51 @@ namespace AudioControls
                 pControl->m_connectedControls = m_connectedControls;
                 pModel->OnControlModified(pControl);
 
+                auto& tmpConnectedControls1 =
+                    connectedControls.size() > m_connectedControls.size() ? connectedControls : m_connectedControls;
+                auto& tmpConnectedControls2 =
+                    connectedControls.size() > m_connectedControls.size() ? m_connectedControls : connectedControls;
+                for (auto& connection1 : tmpConnectedControls1)
+                {
+                    bool bCheck = true;
+                    for (auto& connection2 : tmpConnectedControls2)
+                    {
+                        if (connection1 == connection2)
+                        {
+                            bCheck = false;
+                            break;
+                        }
+                    }
+
+                    if (!bCheck)
+                    {
+                        continue;
+                    }
+
+                    if (IAudioSystemEditor* audioSystemImpl = CAudioControlsEditorPlugin::GetImplementationManager()->GetImplementation())
+                    {
+                        if (IAudioSystemControl* middlewareControl = audioSystemImpl->GetControl(connection1->GetID()))
+                        {
+                            if (connectedControls.size() > m_connectedControls.size())
+                            {
+                                audioSystemImpl->ConnectionRemoved(middlewareControl);
+                                pControl->SignalConnectionRemoved(middlewareControl);
+                            }
+                            else
+                            {
+                                TConnectionPtr connection =
+                                    audioSystemImpl->CreateConnectionToControl(pControl->GetType(), middlewareControl);
+                                if (connection)
+                                {
+                                    pControl->SignalConnectionAdded(middlewareControl);
+                                }
+                            }
+
+                            pControl->SignalControlModified();
```

Comment:
I think this is redundant with the `pModel->OnControlModified(pControl);` line above.  You can probably move that line down below your added logic here.